### DISCUSSION TITLE
Replacement for wc_get_log_file_path?

### DISCUSSION THREAD

**Discussion by webdados** [woocommerce/woocommerce#44369] [General]
### Describe the solution you'd like

Sorry, this is not a bug neither an enhancement request, but I wasn't sure where I could reach Woo developers directly, in a written/organised way.

Since 6.8.0-beta.1 I'm getting the deprecation notice for wc_get_log_file_path.

I'm using this on one of my plugins, a payment gateway, to show the shop owner where the logs are stored.

What's the rationale behind having no alternative? Are the text logs going to be discontinued in favor of only the database logs?

**Comment by tammullen** [comment]
Thanks for your question and feedback @webdados about the logging updates upcoming in 8.6.

As you note this isn't a bug or an enhancement request so I'm going to convert this issue into a discussion and the team working on logging can give you an answer there.

**Comment by coreymckrill** [comment]
Hey @webdados 

> What's the rationale behind having no alternative? Are the text logs going to be discontinued in favor of only the database logs?

I think the rationale was just that this function (and `wc_get_log_file_name`) is old, from a time when there were only text logs, not a database option, and the way they were being used in WC Core was not compatible with that. To be clear, text logs are not going to be discontinued (they are, in fact, the preferred and recommended logging method rather than the database method).

> I'm using this on one of my plugins, a payment gateway, to show the shop owner where the logs are stored.

This is useful to know. I'm curious why you would point them to the location of the file on their server instead of pointing them to the log file viewer at WooCommerce > Status > Logs?

**Reply by nigeljamesstevenson** [reply]
Hi folks, I see a support forum post related to the wc_get_log_file_path function [here](https://wordpress.org/support/topic/wc-8-6-0-deprecated-wc_get_log_file_path-function/) so I have pointed them in the direction of this discussion

**Comment by webdados** [comment]
> This is useful to know. I'm curious why you would point them to the location of the file on their server instead of pointing them to the log file viewer at WooCommerce > Status > Logs?

Because most of the times we need them to send us the log files via email for debugging.

**Reply by coreymckrill** [reply]
FYI, as of WC 8.6, the log file viewing interface now includes a button to download the log file, and you can also download multiple log files at once from the log file list table!

**Comment by zeinnicholas** [comment]
@claudiosanches

It seems that your Correios plugin is also triggering this new warning. The issue stems from the `get_log_link()` function used on the description of the option to enable the log files on all the different shipping methods. It looks like your plugin would need an update.

![image](https://github.com/woocommerce/woocommerce/assets/26891676/fbfbcd33-3ef7-4632-a2fe-be7a9c4b6431)
![image](https://github.com/woocommerce/woocommerce/assets/26891676/66be5d75-b510-4040-8224-6e6d9a539039)

**Reply by webdados** [reply]
A lot of plugins will need updates ;-)

**Reply by claudiosanches** [reply]
Fixed already in the latest version of the Correios plugin.

**Comment by Johanwalter** [comment]
Hi

1- Can someone please confirm that updating websites to WC 8.6.0 is safe in spite of error message
Deprecated: wc_get_log_file_path function is deprecated since version 8.6.0, no alternative is available. in /var/www/vhosts/xxxxx.com/wp-includes/functions.php on line 6031

2- What is impact of deprecated function wc_get_log_file_path ?

Thanks

**Reply by coreymckrill** [reply]
Hi @Johanwalter 

1. Even though the function is deprecated, it still returns a value, so it should not cause any issues when you upgrade.
2. As mentioned [here](https://github.com/woocommerce/woocommerce/discussions/44369#discussioncomment-8494266), log files can now be downloaded from the Logs screen. This deprecation does not impact the logging system itself, it's just to let developers know they are using an obsolete method.

**Comment by NeaMitika** [comment]
On my website on the latest woocommerce update i get this notice while the debug is active:

`Deprecated: La funzione wc_get_log_file_path è deprecata dalla versione 8.6.0 senza alcuna alternativa disponibile. in httpdocs/wp-includes/functions.php on line 6031`

the 6031 line, is the function `wp_trigger_error` that manages the errors.

How do i find where is the function `wc_get_log_file_path`   used?

**Reply by coreymckrill** [reply]
@NeaMitika I believe that message is generated by WordPress itself. It's a bummer that the error location that it shows isn't more helpful. If you are comfortable adding code snippets to your site, you could try something like this (I haven't actually tested it, so use at your own risk!)

```php
add_action(
	'deprecated_function_run',
	function( $function_name ) {
		wc_get_logger()->debug(
			'Deprecated function used.',
			array(
				'source'    => 'wc_deprecated_function',
				'function'  => $function_name,
				'backtrace' => true,
			)
		);
	}
);
```

Then you would get messages on the Logs screen that have a complete backtrace, which would presumably reveal which plugin is using the deprecated function.

**Reply by NeaMitika** [reply]
Thx @coreymckrill, i dont know if it was cuz of your function but after some searching i found a log inside the woocommerce, called "`wc_deprecated_function-2024-02-29`" and there are all the backtrace.

The problem is with the plugin `yith-paypal-express-checkout-for-woocommerce`
on the file `yith-paypal-express-checkout-for-woocommerce\includes\admin\gateway-settings.php`
line `235`
`'description' => sprintf( __( 'Log Paypal Express Checkout events inside <code>%s</code>', 'yith-paypal-express-checkout-for-woocommerce' ), wc_get_log_file_path( 'yith-paypal-ec' ) ),`

My solution is to replace the old line with:
`'description' => sprintf( __( 'Log Paypal Express Checkout events inside <code>%s</code>', 'yith-paypal-express-checkout-for-woocommerce' ), 'wc-logs/yith-paypal-ec.log' ),`

Do you think this solution makes sense?

**Reply by coreymckrill** [reply]
@NeaMitika I guess that will temporarily prevent the deprecation errors from showing up. But if YITH releases a new version of that plugin, your code customization will get overwritten. I would recommend reporting the issue to them so that they can fix it permanently in their next release.

Ideally the message would change to just link to the Logs screen (WooCommerce > Status > Logs) rather than outputting a path on the web server's file system.

**Reply by NeaMitika** [reply]
Yea i tought that too, its fine for me that my change will be overwritten by new update, i think they will fix it too.. 
I tried to search for a repo of the plugin somewere on the web but i couldnt find anything but the the WordPress plugin repository changelog sistem. And i couldnt find any issues section.. 
But its not that important, i m sure they will fix it. 

Thx a lot for the replies!
